audformat/core/utils.py

```diff
@@ -326,6 +326,34 @@ def expand_file_path(
     return index
 
 
+def has_overlap(
```

Comment:
Maybe this is a bit too generic, maybe we should change to `index_has_overlap`? @hagenw what do you think?

Reply:
I agree, let's change it to that so it's more clear

Reply:
Or, why don't we keep the name as it is but add support for pd.Series and pd.Dataframe, just like we did with `iter_by_file` ?

Reply:
When I see `audformat.utils.has_overlap(df)` it is not intuitive for a user to know what will happen. As a user I would probably assume it checks if values in the columns overlap and not in the index. So I would suggest to rename and only support index.

Reply:
If we force the user to do `audformat.utils.index_has_overlap(df.index)` the code will be longer, but it's also clear what the statement does.

Reply:
I added support for `Series` and `Dataframe`, if you think I should revert it to just `pd.Index` let me know

Reply:
Ok, as long as we rename to `index_has_overlap()` it is probably ok (and convenient) to support `pd.Series` and `pd.DataFrame` as we do with other functions. Please extend the tests accordingly.

Reply:
Oh the page didn't reload and didn't see your comments on not adding support for Dataframes and Series. But with the last comment I guess should be fine, I'll just rename it so it's more clear